### PR TITLE
Time map datepicker

### PR DESCRIPTION
**Pull request by jorge-sepulveda**
Went back to the time map component to add a datepicker. Also went ahead and brought it up to stricter standards to keep it consistent with the lockdown map component.

### CODE REVIEW

File: libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts [hunk 1]
```diff
@@ -79,62 +87,67 @@ export class TimeMapComponent implements OnInit {
...
-      '.json';
-    const countyURL =
-      'https://raw.githubusercontent.com/jorge-sepulveda/covid-time-map/master/src/pyscraper/outputFiles/counties/' +
-      currentDateSelected +
-      '.json';
+  public reloadData(): void {
+    this.dateSelected = this.datePicker.nativeElement.value;
+    const stateURL = `https://raw.githubusercontent.com/jorge-sepulveda/covid-time-map/master/src/pyscraper/outputFiles/states/${this.dateSelected}.json`;
+    const countyURL = `https://raw.githubusercontent.com/jorge-sepulveda/covid-time-map/master/src/pyscraper/outputFiles/counties/${this.dateSelected}.json`;
+    this.dateModel.next(this.dateSelected);
```
Comment (Proantagonist) [inline libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts:135]: Refactor. Let the model be the source of truth, and not the view input value.

File: libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts [hunk 2]
```diff
@@ -15,16 +14,26 @@ import { MapboxMapService } from '@tamu-gisc/maps/mapbox';
...
+  @ViewChild('datePicker', { static: false })
+  public datePicker: ElementRef;
+
+  public ngOnInit(): void {
+    this.mapService.loaded.pipe(take(1)).subscribe((map) => {
+      const maxDate: Date = new Date(this.currentDay - 1 * 24 * 60 * 60 * 1000);
+      this.dateSelected = maxDate.toISOString().split('T')[0];
+      this.datePicker.nativeElement.value = this.dateSelected;
+      this.datePicker.nativeElement.max = this.dateSelected;
       const zoomThreshold = 3;
```
[the remark this answers is not in the document]
Comment (jorge-sepulveda) [reply libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts:38]: removed

File: libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.html [hunk 3]
```diff
@@ -4,6 +4,10 @@
       <h3>Map Controls</h3>
       <tamu-gisc-checkbox label="Counties/State" (click)="stateToggle()"></tamu-gisc-checkbox>
       <tamu-gisc-checkbox label="Infection Rate/Mortality Rate" (click)="mortalityToggle()"></tamu-gisc-checkbox>
+      <label for="start">Selected date: </label>
+      <input #datePicker type="date" name="trip-start" min="2020-01-21">
+      <tamu-gisc-button id="dateButton" value="Reload Map" type="submit" (click)="reloadData()"></tamu-gisc-button>
```
[the remark this answers is not in the document]
Comment (jorge-sepulveda) [reply libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.html:9]: changed it and removed the button

[the remark this answers is not in the document]
Comment (jorge-sepulveda) [reply libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.html:9]: removed button and moved input and label into a div.

File: libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts [hunk 4]
```diff
@@ -15,16 +14,26 @@ import { MapboxMapService } from '@tamu-gisc/maps/mapbox';
 export class TimeMapComponent implements OnInit {
   constructor(private mapService: MapboxMapService, private http: HttpClient) {}
 
-  private maxDate: string;
   private stateData: Array<StateRecord>;
   private countyData: Array<CountyRecord>;
-
   public infoBoxModel: BehaviorSubject<IInfoBox> = new BehaviorSubject(undefined);
+  public dateModel: BehaviorSubject<String> = new BehaviorSubject(undefined);
```
[the remark this answers is not in the document]
Comment (jorge-sepulveda) [reply libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts:20]: changed it

File: libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts [hunk 5]
```diff
@@ -15,16 +14,26 @@ import { MapboxMapService } from '@tamu-gisc/maps/mapbox';
...
+
+  public dateSelected: string;
+  public currentDay: number = new Date().getTime();
```
[the remark this answers is not in the document]
Comment (jorge-sepulveda) [reply libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts:23]: moved currentDay into `ngOnInit()`

File: libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts (hunk 1 above)
Comment (jorge-sepulveda) [reply libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts:135]: switched and removed DatePicker as there is no use for it anymore

File: libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jorge-sepulveda) [reply libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts:20]: initialized `dateModel` inside `ngOnInit()` with a string value

File: libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts [hunk 6]
```diff
@@ -15,16 +14,26 @@ import { MapboxMapService } from '@tamu-gisc/maps/mapbox';
...
-    this.mapService.loaded.subscribe((map) => {
+  @ViewChild('datePicker', { static: false })
+  public datePicker: ElementRef;
+
+  public ngOnInit(): void {
+    this.mapService.loaded.pipe(take(1)).subscribe((map) => {
+      const maxDate: Date = new Date(this.currentDay - 1 * 24 * 60 * 60 * 1000);
+      this.dateSelected = maxDate.toISOString().split('T')[0];
+      this.datePicker.nativeElement.value = this.dateSelected;
+      this.datePicker.nativeElement.max = this.dateSelected;
```
[the remark this answers is not in the document]
Comment (jorge-sepulveda) [reply libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts:36]: resolved

File: libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.html [hunk 7]
```diff
@@ -4,6 +4,10 @@
       <h3>Map Controls</h3>
       <tamu-gisc-checkbox label="Counties/State" (click)="stateToggle()"></tamu-gisc-checkbox>
       <tamu-gisc-checkbox label="Infection Rate/Mortality Rate" (click)="mortalityToggle()"></tamu-gisc-checkbox>
+      <label for="start">Selected date: </label>
+      <input #datePicker type="date" name="trip-start" min="2020-01-21">
```
[the remark this answers is not in the document]
Comment (jorge-sepulveda) [reply libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.html:8]: Resolved.

**Pull request merged**